crates/sim2h/src/lib.rs

```diff
@@ -260,49 +231,565 @@ impl Sim2hHandle {
+    debug!("Sending Pong in response to Ping");
+    sim2h_handle.send(signer, uri, &WireMessage::Pong);
+}
+
+fn spawn_handle_message_status(sim2h_handle: Sim2hHandle, uri: Lib3hUri, signer: AgentId) {
+    tokio::task::spawn(async move {
+        debug!("Sending StatusResponse in response to Status");
+        let state = sim2h_handle.im_state().get_clone().await;
+        sim2h_handle.send(
+            signer.clone(),
+            uri.clone(),
+            &WireMessage::StatusResponse(StatusData {
+                spaces: state.spaces_count(),
+                connections: sim2h_handle.connection_count.get().await,
+                redundant_count: match sim2h_handle.dht_algorithm() {
+                    DhtAlgorithm::FullSync => 0,
+                    DhtAlgorithm::NaiveSharding { redundant_count } => *redundant_count,
+                },
+                version: WIRE_VERSION,
+            }),
+        );
+    });
+}
+
+fn spawn_handle_message_hello(
+    sim2h_handle: Sim2hHandle,
+    uri: Lib3hUri,
+    signer: AgentId,
+    version: u32,
+) {
+    /*
+    tokio::task::spawn(async move {
+    });
+    */
+    // no processing here, don't bother actually spawning
+    debug!("Sending HelloResponse in response to Hello({})", version);
+    sim2h_handle.send(
+        signer,
+        uri.clone(),
+        &WireMessage::HelloResponse(HelloData {
+            redundant_count: match sim2h_handle.dht_algorithm() {
+                DhtAlgorithm::FullSync => 0,
+                DhtAlgorithm::NaiveSharding { redundant_count } => *redundant_count,
+            },
+            version: WIRE_VERSION,
+            extra: None,
+        }),
+    );
+    // versions do not match - disconnect them
+    if version != WIRE_VERSION {
+        warn!(
+            "Disconnecting client for bad version this WIRE_VERSIO = {}, client WIRE_VERSION = {}",
+            WIRE_VERSION, version
+        );
+        sim2h_handle.disconnect(vec![uri]);
+    }
+}
+
+fn spawn_handle_message_join_space(
+    sim2h_handle: Sim2hHandle,
+    uri: Lib3hUri,
+    _signer: AgentId,
+    data: SpaceData,
+) {
+    sim2h_handle.im_state().spawn_new_connection(
+        data.space_address.clone(),
+        data.agent_id.clone(),
+        uri.clone(),
+    );
+
+    sim2h_handle.send(
+        data.agent_id.clone(),
+        uri.clone(),
+        &WireMessage::Lib3hToClient(Lib3hToClient::HandleGetGossipingEntryList(GetListData {
+            request_id: "".into(),
+            space_address: data.space_address.clone(),
+            provider_agent_id: data.agent_id.clone(),
+        })),
+    );
+
+    sim2h_handle.send(
+        data.agent_id.clone(),
+        uri,
+        &WireMessage::Lib3hToClient(Lib3hToClient::HandleGetAuthoringEntryList(GetListData {
+            request_id: "".into(),
+            space_address: data.space_address.clone(),
+            provider_agent_id: data.agent_id,
+        })),
+    );
+}
+
+fn spawn_handle_message_send_dm(
```

Comment:
The only difference between `spawn_handle_message_send_dm` and `spawn_handle_message_send_dm_result` is the enum flavor `SendDirectMessageResult` or `SendDirectMessage`  
This seems worth refactoring?